### ISSUE TITLE
JWT Verify: it re-downloads the JWKS for any request with a jwt token that has an invalid kid

### ISSUE THREAD

**Issue by abrassara** [zmartzone/lua-resty-openidc#468]
###### Environment

- lua-resty-openidc version 1.7.6-3

###### Actual behaviour

When verifying JWT tokens, if the kid of the JWT is not found in the cached JWKS, it re-downloads the JWKS and checks the kid against the downloaded version.

###### Expected behaviour

It should use the information in the cached JWKS and, if the kid is not found, the validation should directly fail. The information in the JWKS rarely changes. If an attempt is made to download the JWKS for any request that has a token with an invalid kid, it adds penalty to these requests since it depends on the response time of the JWKS source. Also it could overload the JWKS source and be prone to ddos ​​attacks, for example.

**Comment by bodewig** [comment]
It seems this is intentional, #108 introduced the cache and at the same time made sure the cache is bypassed if a key has not been found in order to support JWK rotations.

We could introduce an optional flag to suppress the second cache lookup but would likely need to keep the existing behavior as default.

**Comment by abrassara** [comment]
Yes, it would be wise to keep the default behavior but also have an option to disable it and avoid the second cache lookup that results in downloading JWKS.
